### ISSUE TITLE
The back button does not always work on ZIM files

### ISSUE THREAD

**Issue by mossroy** [kiwix/kiwix-html5#177]
The back button of the browser (or the left arrow in the bottom bar of the screen) should allow the user to browse his previous pages.
It works fine on Evopedia archives, but not always on ZIM files

**Comment by sharun-s** [comment]
".html" is missing from the titleName argument passed to pushBrowserHistoryState function.

When title search OR the random article button is pressed pushBrowserHistoryState(title.name()) is called. In contrast, when a link in a displayed article is clicked pushBrowserHistoryState(decodeURIComponent(url)) is called which has the ".html". So history works when article links are being clicked, and fails when RandomArticle button is click or a Search result link is clicked. So depending on the *sequence* of user clicks history state can get messed up sometimes and not others. If the user is just clicking through links in the article history will work fine.

The "hack" fix would be to change pushBrowserHistoryState(title.name()) to pushBrowserHistoryState(title.name() + ".html"); in handleTitleName and gotoRandomArticle. Back button will start working.

A real fix would require understanding of how url's are mapping to articles in the zim. Not clear to me whats going on. Initially I noticed if I cut and paste a article url in a new tab it doesn't work or use right click open link in new tab. This I assume is about maintaining session state between tabs. 
But then I found an article link url and and pasting in the same tab that has already loaded/displayed an article doesn't work either. So it feels like fixing history and the back button is a secondary issue until this url's to an article mapping is clearer.
